node/engine/messageservice/p2p-message-service/dht-record.go

```diff
@@ -0,0 +1,85 @@
+package p2pms
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+const (
+	DHT_RECORD_PREFIX = "/scaddr/"
+)
+
+type stateChannelAddrToPeerIDValidator struct{}
+
+// DhtRecord represents the data stored in the DHT record
+type DhtRecord struct {
+	Data      DhtData `json:"data"`
+	Signature []byte  `json:"signature"`
+}
+
+type DhtData struct {
+	SCAddr    string `json:"scaddr"` // state channel address
+	PeerID    string `json:"peerid"`
+	Timestamp int64  `json:"timestamp"` // Unix timestamp (seconds since January 1, 1970)
+}
+
+func (v stateChannelAddrToPeerIDValidator) Validate(key string, value []byte) error {
+	// Trim the DHT_RECORD_PREFIX from the key to get the state channel address
+	signingAddrStr := strings.TrimPrefix(key, DHT_RECORD_PREFIX)
+
+	// Check if it's a valid state channel address
+	if !common.IsHexAddress(signingAddrStr) {
+		return errors.New("invalid state channel address used for key")
+	}
+
+	// Parse the value into a RecordData object
+	var dhtRecord DhtRecord
+	if err := json.Unmarshal(value, &dhtRecord); err != nil {
+		return errors.New("malformed record value")
+	}
+
+	// Make sure the timestamp is not in the future or negative number
+	if dhtRecord.Data.Timestamp > time.Time.Unix(time.Now()) || dhtRecord.Data.Timestamp < 0 {
```

Comment:
Could this cause problems due to differing clocks on different machines? The clock on my machine could be ahead of the clock on your machine.

Reply:
Thats a good point. Do you think I should just remove this timestamp check from the `Validate` method? I'm not sure that its necessary.